### PR TITLE
implement getAllValues feature

### PR DESCRIPTION
**Pull request by endret**

### CODE REVIEW

**Review by laliconfigcat — APPROVED** [review]

File: src/ConfigCatClient.ts [hunk 1]
```diff
@@ -221,6 +227,36 @@ export class ConfigCatClient implements IConfigCatClient {
...
+        this.getAllValuesAsync(user).then(value => {
+            callback(value);
+        });
+    }
+    
+    getAllValuesAsync(user?: User): Promise<SettingKeyValue[]> {
+        return new Promise(async (resolve) => {
+            const config = await this.configService.getConfig();
+            if (!config || !config.ConfigJSON || !config.ConfigJSON[ConfigFile.FeatureFlags]) {
+                this.options.logger.error("JSONConfig is not present, returning empty array");
```
Comment (sigewuzhere) [inline src/ConfigCatClient.ts]: Suggestion for this and other similar error messages:
`config.json is not present in the cache, returning empty array`
or
`getAllValuesAsync(): config.json is not present in the cache, returning empty array`

File: src/ConfigCatClient.ts [hunk 2]
```diff
@@ -47,6 +47,12 @@ export interface IConfigCatClient {
 
     /** Releases all resources used by IConfigCatClient */
     dispose(): void;
+
+    /** Returns all values of a feature flag or setting */
```
Comment (sigewuzhere) [inline src/ConfigCatClient.ts:51]: `Evaluates and returns the values of all feature flags and settings`

File: src/ConfigCatClient.ts [hunk 3]
```diff
@@ -47,6 +47,12 @@ export interface IConfigCatClient {
...
+    getAllValues(callback: (result: SettingKeyValue[]) => void, user?: User): void;
+
+    /** Returns all values of a feature flag or setting */
```
Comment (sigewuzhere) [inline src/ConfigCatClient.ts:54]: same

**Review by sigewuzhere — COMMENTED** [review]
Found two misleading comments

**Pull request merged**